docs/data/tree-view/rich-tree-view/headless/headless.md

```diff
@@ -0,0 +1,269 @@
+---
+productId: x-tree-view
+title: Rich Tree View - Headless
+packageName: '@mui/x-tree-view'
+githubLabel: 'component: tree view'
+waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
+---
+
+# Rich Tree View - Headless
+
+<p class="description">Create your custom tree view.</p>
+
+:::warning
+The `useTreeView` hook is not public API for now,
+this page is still a _Work in Progress_ and its content is not meant to be used in production.
+:::
+
+## Create a custom plugin
+
+### Basic plugin
+
+A custom plugins contains 2 required elements:
+
+1. A hook that will be executed by the `useTreeView` hook:
+
+   ```ts
+   const useCustomPlugin = ({ params }) => {
+     React.useEffect(() => {
+       console.log(params.customParam);
+     });
+   };
+   ```
+
+2. A property containing the params used by this topic:
+
+   ```ts
+   useCustomPlugin.params = {
+     customParam: true,
+   };
+   ```
+
+### Params default value
+
+You can use the `getDefaultizedParams` property to set a default value to your plugin params:
+
+```ts
+const useCustomPlugin = ({ params }) => {
+  React.useEffect(() => {
+    console.log(params.customParam);
+  });
+};
+
+useCustomPlugin.params = { customParam: true };
+
+useCustomPlugin.getDefaultizedParams = (params) => ({
+  ...params,
+  customParam: params.customParam ?? false,
+});
+```
+
+### Controllable models
+
+A model is a value that can either be controlled or initialized using a prop.
+The Tree View contains several models like the `expandedNodes` model which contains the id of the nodes currently expanded.
+
+You can create your own models using the `models` property of your plugin:
+
+```ts
+useCustomPlugin.params = {
+  // ...other params
+  defaultCustomModel: true,
+  customModel: true,
+};
+
+useCustomPlugin.getDefaultizedParams = (params) => ({
+  ...params,
+  // ... other defaultized params
+  defaultCustomModel: params.defaultCustomModel ?? false,
+});
+
+useCustomPlugin.models = {
+  customModel: {
+    getDefaultValue: (params) => defaultCustomModel,
+  },
+};
+```
+
+:::info
+When creating a model, you should always set a default value to your `defaultXXX` prop.
+See the [Params default value](/x/react-tree-view/rich-tree-view/headless/#params-default-value) section for more information.
+:::
+
+You can then use this model in your plugin (or in any other plugin) using the `models` parameter:
+
+```ts
+const useCustomPlugin = ({ models }) => {
+  React.useEffect(() => {
+    console.log(models.customModel.value);
+  });
+
+  const updateCustomModel = (newValue) =>
+    models.customModel.setControlledValue(newValue);
```

Comment:
Should we mention that each model contains a value and an updater?

Reply:
I can do a follow up diving deeper on the model part :ok_hand: